src/overlays/actors/ovl_Obj_Oshihiki/z_obj_oshihiki.c

```diff
@@ -27,63 +67,661 @@ const ActorInit Obj_Oshihiki_InitVars = {
+    DynaPolyActor* dyna;
+
+    if ((this->floorBgIds[this->highestFloor] != 0x32 /*BGCHECK_SCENE*/) &&
+        (fabsf(this->dyna.actor.groundY - this->dyna.actor.posRot.pos.y) < 0.001f)) {
+        dyna = DynaPolyInfo_GetActor(&globalCtx->colCtx, this->floorBgIds[this->highestFloor]);
+        if ((dyna != NULL) && (dyna->actor.id == ACTOR_OBJ_OSHIHIKI)) {
+            return (ObjOshihiki*)dyna;
+        }
+    }
+    return NULL;
+}
+
+void ObjOshihiki_UpdateInitPos(ObjOshihiki* this) {
+    if (this->dyna.actor.initPosRot.pos.x < this->dyna.actor.posRot.pos.x) {
+        while ((this->dyna.actor.posRot.pos.x - this->dyna.actor.initPosRot.pos.x) >= 20.0f) {
+            this->dyna.actor.initPosRot.pos.x += 20.0f;
+        }
+    } else {
+        while ((this->dyna.actor.initPosRot.pos.x - this->dyna.actor.posRot.pos.x) >= 20.0f) {
+            this->dyna.actor.initPosRot.pos.x -= 20.0f;
+        }
+    }
+    if (this->dyna.actor.initPosRot.pos.z < this->dyna.actor.posRot.pos.z) {
+        while ((this->dyna.actor.posRot.pos.z - this->dyna.actor.initPosRot.pos.z) >= 20.0f) {
+            this->dyna.actor.initPosRot.pos.z += 20.0f;
+        }
+    } else {
+        while ((this->dyna.actor.initPosRot.pos.z - this->dyna.actor.posRot.pos.z) >= 20.0f) {
+            this->dyna.actor.initPosRot.pos.z -= 20.0f;
+        }
+    }
+}
+
+s32 ObjOshihiki_NoSwitchPress(ObjOshihiki* this, DynaPolyActor* dyna, GlobalContext* globalCtx) {
+    s16 dynaSwitchFlag;
+
+    if (dyna == NULL) {
+        return 1;
+    } else if (dyna->actor.id == ACTOR_OBJ_SWITCH) {
+        dynaSwitchFlag = (dyna->actor.params >> 8) & 0x3F;
+        switch (dyna->actor.params & 0x33) {
+            case 0x20: // (OBJSWITCH_SUBTYPE_FLOOR_2 << 4) | OBJSWITCH_TYPE_FLOOR
+                if ((dynaSwitchFlag == ((this->dyna.actor.params >> 8) & 0x3F)) &&
+                    Flags_GetSwitch(globalCtx, dynaSwitchFlag)) {
+                    return 0;
+                }
+                break;
+            case 0x30: // (OBJSWITCH_SUBTYPE_FLOOR_3 << 4) | OBJSWITCH_TYPE_FLOOR
+                if ((dynaSwitchFlag == ((this->dyna.actor.params >> 8) & 0x3F)) &&
+                    !Flags_GetSwitch(globalCtx, dynaSwitchFlag)) {
+                    return 0;
+                }
+                break;
+        }
+    }
+    return 1;
+}
+
+void ObjOshihiki_CheckType(ObjOshihiki* this, GlobalContext* globalCtx) {
+    switch (BLOCK_TYPE) {
+        case SMALL_SWITCH_OFF:
+        case MEDIUM_SWITCH_OFF:
+        case LARGE_SWITCH_OFF:
+        case HUGE_SWITCH_OFF:
+        case SMALL_SWITCH_ON:
+        case MEDIUM_SWITCH_ON:
+        case LARGE_SWITCH_ON:
+        case HUGE_SWITCH_ON:
+            ObjOshihiki_InitDynapoly(this, globalCtx, &D_05004E98, 1);
+            return;
+    }
+    // Error : type cannot be determined
+    osSyncPrintf("Error : タイプが判別できない(%s %d)(arg_data 0x%04x)\n", "../z_obj_oshihiki.c", 444,
```

Comment:
This should probably be a default instead of the other case being an early return.